android/src/main/java/com/reactnativelocalserver/tcp/TCPServer.java

```diff
@@ -128,7 +134,10 @@ public void closeConnection(String connectionId, String reason) throws Exception
 
     private void cleanUp(String reason) {
         Log.d(TAG, "clean up: " + id);
-        if (nsdManager != null) {
+        // This way we make sure that DNSSD is used instead of the NSDManager
+        if (dnssd != null) {
+            discovery.unregister();
```

Comment:
This is weird to have unregister() without params when we can switch between DNSSD and NSDManager